File: 1147-longest-chunked-palindrome-decomposition/1147-longest-chunked-palindrome-decomposition.js
/**
 * @param {string} text
 * @return {number}
 */
var longestDecomposition = function(text) {
  if (!text) return 0;
  
  for (let i = 0; i <= text.length / 2; i++) {
    if (text.slice(0, i) === text.slice(-i)) 
      return 2 + longestDecomposition(text.slice(i, -i));
  }
  
  return 1;
};

File: 1147-longest-chunked-palindrome-decomposition/1147-longest-chunked-palindrome-decomposition.py
class Solution:
    def longestDecomposition(self, text: str) -> int:
        ans = 0
        while text:
            i = 1
            while i <= len(text) // 2 and text[:i] != text[-i:]:
                i += 1
            if i > len(text) // 2:
                ans += 1
                break
            ans += 2
            text = text[i:-i]
        return ans